### PR TITLE
Attempt to speed up container builds

### PR DESCRIPTION
**Pull request by blt**
### What does this PR do?

The lading container build right now is roughly 1 hour. This is a very long loop to wait on. Introduce cargo-chef and more aggressive caching to hopefully cut this time.

### CODE REVIEW

**Comment by GeorgeHahn** [comment]
As a smoke test, could you push a small code change?

File: Dockerfile [hunk 1]
```diff
@@ -1,18 +1,36 @@
...
+# Lading arm64 builds are very slow. We use cargo-chef and aggressive layer
+# caching to avoid some of the build delay.
 
+# Builder image, cargo-chef installed
+FROM docker.io/rust:1.81.0-bullseye AS chef
+WORKDIR /app
 RUN apt-get update && apt-get install -y \
     protobuf-compiler fuse3 libfuse3-dev \
     && rm -rf /var/lib/apt/lists/*
+RUN cargo install cargo-chef
```
Comment (GeorgeHahn) [inline Dockerfile:10]: This should probably use the project lockfile, `--locked`

Comment (blt) [reply Dockerfile]: Addressed in [759c7c8](https://github.com/DataDog/lading/pull/1088/commits/759c7c82d4945df7b4521c38ce17cd70ed0fe28e)

**Comment by scottopell** [comment]
Does stripping the release binary help with compile time? I think its useful to preserve debug symbols as a general practice, unless there's a compelling reason.

**Comment by blt** [comment]
> Does stripping the release binary help with compile time? I think its useful to preserve debug symbols as a general practice, unless there's a compelling reason.

Yeah it shaves ~4 minutes off. The lading binary has gotten fairly big. 

On the whole, I'm meh on this change I think. It complicates the build non-trivially and it doesn't really improve build speed. An ARM build machine is what we need.

**Pull request closed without merging**